packages/keplr/src/extension/chain-wallet.ts

```diff
@@ -28,7 +31,33 @@ export class ChainKeplrExtension extends ChainWalletBase<
       if (!keplr) {
         throw new Error('No Keplr Client found!');
       }
-      const key = await keplr.getKey(this.chainName);
+
+      const suggestChain = chainRegistryChainToKeplr(
+        this.chain,
+        [this.assetList],
+        {
+          getRestEndpoint: (chain: Chain) => {
+            if (preferredEndpoints[chain.chain_name]) {
+              return preferredEndpoints[chain.chain_name].rest[0];
+            }
+          },
+          getRpcEndpoint: (chain: Chain) => {
+            if (preferredEndpoints[chain.chain_name]) {
+              return preferredEndpoints[chain.chain_name].rpc[0];
+            }
+          },
+          getExplorer: (chain: Chain) => {
+            return chain.explorers?.[0]?.url;
+          },
+        }
+      );
+
+      console.log(suggestChain);
+
+      await keplr.experimentalSuggestChain(suggestChain);
```

Comment:
I tried doing this in `update()` but I don't think it's the right place. cc @liujun93